### ISSUE TITLE
MacOS_SDK needs updating from 10.10

### ISSUE THREAD

**Issue by ChrisJefferson** [NixOS/nixpkgs#15718]
## Issue description

`os-specific/darwin/apple-sdk` hard-wires the SDK version to 10.9, but `os-specific/darwin/apple-source-releases` includes framework versions from 10.11 and 10.10, including `IOKitUser`.

The `IOKitUser` then won't work in 10.11, as it includes strings like `AVAILABLE_MAC_OS_X_VERSION_10_10_AND_LATER` which should be defined in apple-sdk, but aren't defined in 10.9's (obviously).

So basically, apple-sdk needs versioning in the same was as the frameworks.
### Steps to reproduce

I found this trying to make a working mplayer package, but anything which tries to `#include "/hid/IOHIDDevice.h"` from IOKit will hit this problem.
## Technical details
- System: Mac os X 10.11.5
- Nix version: 1.11.2
- Nixpkgs version: "16.09pre83147.df89584"

**Comment by copumpkin** [comment]
I've started the work to update the whole darwin shebang, but it's a decent amount of work and I'm short on time. I doubt it makes much sense to hand it off in its current state to anyone else, but when I get finished I'll write up the various gotchas so that this isn't so hard for others to pick up next time around.

**Comment by matthewbauer** [comment]
related to #13779 just so it doesn't get lost

**Comment by matthewbauer** [comment]
I think this can now be closed.

**Comment by aflatter** [comment]
@matthewbauer what has changed?

**Comment by copumpkin** [comment]
Unless someone else has updated it, it still needs a bump. I'll try to get to it soon

**Comment by domenkozar** [comment]
@copumpkin will the shebang refactoring remove MacOS_SDK and OSXPrivateSDK from closures?

**Comment by copumpkin** [comment]
It should eventually. I actually made an attempt to bump this up to 10.11 (we got a native `xhyve` build for a couple of git revisions!!) but it got reverted because it broke a bunch of stuff unexpectedly and I haven't had time to investigate. It should be coming back soon though 😄

**Comment by matthewbauer** [comment]
So what's the status on this issue?

**Comment by copumpkin** [comment]
It's on 10.10 so is technically updated now. I wanted 10.11 but that one is trickier

**Comment by matthewbauer** [comment]
see #37407
